CMakeLists.txt

```diff
@@ -0,0 +1,826 @@
+  # /RTC1 Run-time error checks:
+  add_compile_options($<$<OR:$<CONFIG:Debug>,$<CONFIG:DebugClassic>>:/RTC1>)
+
+# Generator expressions are not supported in add_definitions, use set_property instead
+set_property(
+  DIRECTORY
+  APPEND
+  PROPERTY COMPILE_DEFINITIONS
+  $<$<OR:$<CONFIG:Debug>,$<CONFIG:DebugClassic>>:_CRTDBG_MAP_ALLOC>)
+
+  # /MT Language: Multi-threaded CRT
+  add_compile_options($<$<OR:$<CONFIG:Release>,$<CONFIG:ReleaseClassic>>:/MT>)
+  add_compile_options($<$<OR:$<CONFIG:Release>,$<CONFIG:ReleaseClassic>>:/Ox>)
+  # /Ox Optimization: Full
+
+endif (NOT WIN32)
+
+############################################################
+
+if (is_clang)
+  set(rocks_db_system_header --system-header-prefix=rocksdb2)
+else()
+  unset(rocks_db_system_header)
+endif()
+
+set(soci_extra_includes
+  -I/"${CMAKE_SOURCE_DIR}/"/src/soci/src/core
+  -I/"${CMAKE_SOURCE_DIR}/"/src/soci/include/private
+  -I/"${CMAKE_SOURCE_DIR}/"/src/sqlite)
+
+if (WIN32 OR is_xcode OR unity)
+  prepend(beast_unity_srcs
+    src/ripple/beast/unity/
+    beast_clock_unity.cpp
+    beast_container_unity.cpp
+    beast_insight_unity.cpp
+    beast_net_unity.cpp
+    beast_nudb_unity.cpp
+    beast_utility_unity.cpp)
+
+  prepend(ripple_unity_srcs
+    src/ripple/unity/
+    app_ledger.cpp
+    app_main.cpp
+    app_misc.cpp
+    app_paths.cpp
+    app_tests.cpp
+    app_tx.cpp
+    core.cpp
+    basics.cpp
+    crypto.cpp
+    ledger.cpp
+    net.cpp
+    overlay.cpp
+    peerfinder.cpp
+    json.cpp
+    protocol.cpp
+    rpcx.cpp
+    shamap.cpp
+    server.cpp
+    test.cpp)
+
+  list(APPEND src ${beast_unity_srcs} ${ripple_unity_srcs})
+
+  add_with_props(src/ripple/unity/nodestore.cpp
+    -I/"${CMAKE_SOURCE_DIR}/"/src/rocksdb2/include
+    -I/"${CMAKE_SOURCE_DIR}/"/src/snappy/snappy
+    -I/"${CMAKE_SOURCE_DIR}/"/src/snappy/config
+    ${rocks_db_system_header})
+
+  add_with_props(src/ripple/unity/soci_ripple.cpp ${soci_extra_includes})
+
+  set(unity_srcs ${beast_unity_srcs} ${ripple_unity_srcs}
+    src/ripple/unity/nodestore.cpp
+    src/ripple/unity/soci_ripple.cpp)
+
+  set_property(
+    SOURCE ${unity_srcs}
+    APPEND
+    PROPERTY HEADER_FILE_ONLY
+    ${nonunity})
+  # Doesn't work
+  # $<OR:$<CONFIG:DebugClassic>,$<CONFIG:ReleaseClassic>>)
+
+endif ()
+
+if (WIN32 OR is_xcode OR NOT unity)
+  file(GLOB_RECURSE core_srcs src/ripple/core/*.cpp)
+  add_with_props("${core_srcs}"
+    -I/"${CMAKE_SOURCE_DIR}/"/src/soci/src/core
+    -I/"${CMAKE_SOURCE_DIR}/"/src/sqlite)
+
+  set(non_unity_srcs ${core_srcs})
+
+  foreach(curdir
+      beast/clock beast/container beast/insight beast/net beast/nudb beast/utility
+      app basics crypto json ledger legacy net overlay peerfinder protocol rpc
+      shamap server test)
+    file(GLOB_RECURSE cursrcs src/ripple/${curdir}/*.cpp)
+    list(APPEND src "${cursrcs}")
+    list(APPEND non_unity_srcs "${cursrcs}")
+  endforeach()
+
+  file(GLOB_RECURSE nodestore_srcs src/ripple/nodestore/*.cpp)
```

Comment:
I have a strong opinion - having to manually edit the CMakeLists.txt for every change in the sources is a non-starter.